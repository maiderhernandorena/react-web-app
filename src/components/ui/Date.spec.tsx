import React from 'react';
import Renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

import DateComponent from './Date';

describe('Date Component', () => {
  describe('Validate props', () => {

    it('Should render correctly with string format', () => {
      const renderer = Renderer.create(<DateComponent date='2020-10-19' />);
      const output = renderer.root;

      expect(output.props.date).toStrictEqual('2020-10-19');
    });    
  });

  describe('Snapshots', () => {
    it('Should render correctly with string format', () => {
      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <DateComponent date='2020-10-19' />
      );
      expect(wrapper).toMatchSnapshot();
    });
    
    it('Should render correctly with string format', () => {
      const date = new Date('2020-10-19')

      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <DateComponent date={date} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});