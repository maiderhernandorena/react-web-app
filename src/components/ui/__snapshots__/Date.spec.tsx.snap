// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Component Snapshots Should render correctly with string format 1`] = `
<p
  className="App-date"
>
  2020-10-19
</p>
`;

exports[`Date Component Snapshots Should render correctly with string format 2`] = `
<p
  className="App-date"
>
  2020-10-19T00:00:00.000Z
</p>
`;
